AutoRest/AutoRest.Core/Utilities/Extensions.cs

```diff
@@ -263,5 +263,54 @@ public static bool IsPrimaryType(this IType type, KnownPrimaryType typeToMatch)
             }
             return false;
         }
+
```

Comment:
@markcowl Mark, could you take a quick look at the changes I made in this last commit whenever it's convenient for you?

For context, MethodTemplateModel.GetSerializationSettingsReference() is getting too complex and failing a build rule for code complexity (and it's hard to read). I simplified this logic by adding methods to check if an IType either is a KnownPrimaryType or is a dictionary/sequence that is based on the KnownPrimaryType (previously these checks were repeated for each primary type that had special serialization behavior).

Tests are passing and the serialization is working the same as before. I'd like another set of eyes on it before I merge to make sure I kept the behavior the same with respect to the null checks. Thanks!